### DISCUSSION TITLE
Reading more than 100 rows of the database

### DISCUSSION THREAD

**Discussion by edelveiss** [orlowdev/gatsby-source-notion-api#5] [Q&A]
Currently it is only possible to read 100 rows of the database. Would be great if there was an option that could be set to any number. Thanks!

**Comment by orlowdev** [comment]
Hi, good point. I'll try to find some time today to add support for recursive database fetching. Thanks!

**Comment by orlowdev** [comment]
@edelveiss I added recursive DB fetching but I couldn't test it out. Please let me know if it worked. The new code is available in https://github.com/orlowdev/gatsby-source-notion-api/releases/tag/0.9.0.

**Comment by edelveiss** [comment]
@orlowdev I really appreciate such a quick response! Do you think you could test it? Because it’s still not working on my side and I am still getting only the first 100 rows.

**Reply by orlowdev** [reply]
OK, my bad, I put the `return` in the wrong place, haha 😅 I updated the code, v0.9.1 is no out. Here's a link that pulls 115 items from a DB successfully: https://over-a-hundred-rows-gatsby-source-notion-api.netlify.app

**Reply by edelveiss** [reply]
@orlowdev thank you so much for the update! Seems to be working now on the existing more than 100 entries. And how you have mentioned it does not work yet for newly created rows.

**Comment by orlowdev** [comment]
There's a weird thing happening. I had this table for blog posts that I used for my blog, and it currently has just 12 entries. I added another 90+ to test it out, but the Notion API only returns me the 12 rows I had initially. Probably some kind of caching is applied. 🤷 Is there a chance you could try fetching your database contents with Postman or any other way, and see what the database returns? If it's ok for you, here's an API reference for querying a database: https://developers.notion.com/reference/post-database-query

Meanwhile, I'll try to test it out with a separate database and see if it works.
